### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by kcharwood**
Merging the 3.0.0 branch to master.

### CODE REVIEW

**Comment by kcharwood** [comment]
@kylef @0xced @cnoon @jshier

I'd love a santity check review here. May be easier to pull the branch and just run `git difftool master..HEAD AFOAuth2Manager/*` since there is so much more project configuration noise migrating this to a framework based approach.

**Comment by jameshays** [comment]
Do we have an ETA on when this will be merged?

**Comment by kcharwood** [comment]
My hope is soon. I've been out on vacation the last two weeks, and I'm at CES this week, so I haven't been able to close it out just yet. My goal is next week.

**Comment by kcharwood** [comment]
If you need to pull this in via CocoaPods, you should be able to do something like:

```
pod 'AFOAuth2Manager', :git => 'https://github.com/AFNetworking/AFOAuth2Manager.git', :branch => '3_0_0_branch'
```

**Comment by jameshays** [comment]
Thanks @kcharwood.  I'm building an extension off of this and will pull that branch in if it's that close to final merge.  Thanks for your work on this!

**Comment by tblanchard** [comment]
Are we there yet?

File: AFOAuth2Manager/AFOAuthCredential.m [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+@interface AFOAuthCredential()
+@property (readwrite, nonatomic, copy) NSString *accessToken;
+@property (readwrite, nonatomic, copy) NSString *tokenType;
+@property (readwrite, nonatomic, copy) NSString *refreshToken;
+@property (readwrite, nonatomic, copy) NSDate *expiration;
+@end
+
+
+@implementation AFOAuthCredential
+//@dynamic expired;
```
Comment (ejensen) [inline AFOAuth2Manager/AFOAuthCredential.m]: This commented-out line should probably be removed

File: AFOAuth2Manager/AFOAuth2Manager.h [hunk 2]
```diff
@@ -18,22 +18,17 @@
...
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// THE SOFTWARE
 
 #import <Foundation/Foundation.h>
+#import "AFOAuthCredential.h"
+#import "AFHTTPRequestSerializer+OAuth2.h"
 
-#import <AFNetworking/AFHTTPRequestOperationManager.h>
+@import AFNetworking;
```
Comment (ejensen) [inline AFOAuth2Manager/AFOAuth2Manager.h]: Can this be switched back to `#import <AFNetworking/AFHTTPSessionManager.h>` in order to maintain compatibility with CocoaPods setups that don't use dynamic frameworks? 

 The change was made in commit 42744f9ce61e12dc641867072bc8d79e145659ec when adding a test target. It's not clear to me why the change was required. The tests complete with the `#import <>` style import.

Somewhat related to issue https://github.com/AFNetworking/AFOAuth2Manager/pull/114

Comment (crsantos) [inline AFOAuth2Manager/AFOAuth2Manager.h]: I second that

**Comment by TiagoVeloso** [comment]
Any news?

**Comment by dstranz** [comment]
@kcharwood Why you are not merging it?

**Comment by kcharwood** [comment]
Merging this into a master! I'll let it simmer for 24 hours before pushing an official release

**Pull request merged**